### PR TITLE
SNOW-1359379:Add VectorType to snowpark python API doc

### PR DESCRIPTION
**Pull request by sfc-gh-yuwang**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes SNOW-1359379

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

**Review by sfc-gh-rpanchapakesan — APPROVED** [review]
Thanks!

**Review by sfc-gh-mloring — APPROVED** [review]

File: docs/README.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ make html
 If you get warnings, like "WARNING: [autosummary] failed to import 'snowpark.dataframe': no module named snowpark.dataframe", try reinstalling the Snowpark API:
 
 ```bash
-python -m pip install ".[development, pandas]"
+python -m pip install ".[development, pandas, modin]"
```
Comment (sfc-gh-aalam) [inline docs/README.md:15]: nit: this looks unrelated to vector type changes

File: src/snowflake/snowpark/functions.py [hunk 2]
```diff
@@ -5793,9 +5793,9 @@ def vector_l2_distance(v1: ColumnOrName, v2: ColumnOrName) -> Column:
     """Returns the cosine distance between two vectors of equal dimension and element type.
```
Comment (sfc-gh-ssung) [inline src/snowflake/snowpark/functions.py:5793]: should this be L2 distance?

File: docs/README.md (hunk 1 above)
Comment (sfc-gh-yuwang) [reply docs/README.md:15]: This is a bug we found, since we have included modin, the doc generator need modin dependencies to work

File: src/snowflake/snowpark/functions.py (hunk 2 above)
Comment (sfc-gh-yuwang) [reply src/snowflake/snowpark/functions.py:5793]: I guess so, let me change it

**Review by sfc-gh-jrose — APPROVED** [review]

File: docs/README.md (hunk 1 above)
Comment (sfc-gh-yixie) [reply docs/README.md:15]: Better use a different PR to address it.

File: src/snowflake/snowpark/functions.py [hunk 3]
```diff
@@ -5775,32 +5775,34 @@ def vector_cosine_distance(v1: ColumnOrName, v2: ColumnOrName) -> Column:
...
 
 def vector_l2_distance(v1: ColumnOrName, v2: ColumnOrName) -> Column:
-    """Returns the cosine distance between two vectors of equal dimension and element type.
+    """Returns the l2 distance between two vectors of equal dimension and element type.
 
     Example::
-        >> from snowflake.snowpark.functions import vector_l2_distance
-        >> df = session.sql("select [1,2,3]::vector(int,3) as a, [2,3,4]::vector(int,3) as b")
-        >> df.select(vector_l2_distance(df.a, df.b).as_("dist")).show()
+        >>> from snowflake.snowpark.functions import vector_l2_distance
```
Comment (sfc-gh-yixie) [inline src/snowflake/snowpark/functions.py:5797]: For my learning, what's the purpose of three >?

Comment (sfc-gh-yuwang) [reply src/snowflake/snowpark/functions.py:5797]: it makes it an actual doc string example code, three > give those lines code block style in the doc

**Comment by sfc-gh-yuwang** [comment]
I had another PR to update the README: https://github.com/snowflakedb/snowpark-python/pull/1491
